### PR TITLE
[enhancement](statistics) collect table level loaded rows on BE to make RPC light weight

### PR DESCRIPTION
**Pull request by Hastyshell**
## Proposed changes

Original way is to collect loaded rows on tablet level on each BE and aggregate them in master FE. For light weight RPC, it is better to do the aggregation on BE.

Result:

```
2023-09-20 16:53:50,761 DEBUG (PUBLISH_VERSION|23) [DatabaseTransactionMgr.updateCatalogAfterVisible():1793] table id to loaded rows:{10325=20}
```

```
mysql> select count(*) from dup_tbl_basic;
+----------+
| count(*) |
+----------+
|       20 |
+----------+
1 row in set (0.05 sec)
```

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run compile

**Comment by morrySnow** [comment]
run p0

**Comment by TangSiyang2001** [comment]
run p1

**Review by freemandealer — APPROVED** [review]
LGTM

File: be/src/olap/task/engine_publish_version_task.cpp [hunk 1]
```diff
@@ -189,8 +189,11 @@ Status EnginePublishVersionTask::finish() {
                 }
             }
             auto rowset_meta_ptr = rowset->rowset_meta();
-            _tablet_id_to_num_delta_rows->insert(
-                    {rowset_meta_ptr->tablet_id(), rowset_meta_ptr->num_rows()});
+            auto table_id = StorageEngine::instance()
+                                    ->tablet_manager()
+                                    ->get_tablet(rowset_meta_ptr->tablet_id())
+                                    ->get_table_id();
+            (*_table_id_to_num_delta_rows)[table_id] += rowset_meta_ptr->num_rows();
```
Comment (zhannngchen) [inline be/src/olap/task/engine_publish_version_task.cpp:196]: You should not accumulate the total rows here, publish tasks might fail and retry, you'd better to move such code to L232

**Comment by TangSiyang2001** [comment]
run buildall

File: be/src/olap/task/engine_publish_version_task.cpp (hunk 1 above)
Comment (TangSiyang2001) [reply be/src/olap/task/engine_publish_version_task.cpp]: Modified.

**Review by zhannngchen — DISMISSED** [review]
LGTM

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run p0

**Comment by TangSiyang2001** [comment]
run p0

**Comment by TangSiyang2001** [comment]
run arm

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run buildall

**Review by zhannngchen — APPROVED** [review]
LGTM

**Pull request merged**